internal/pkg/cli/task_run.go

```diff
@@ -297,14 +310,90 @@ func (o *runTaskOpts) Execute() error {
 		}
 	}
 
-	_, err := o.runTask()
+	tasks, err := o.runTask()
 	if err != nil {
 		return err
 	}
 
+	if o.follow {
+		if err := o.displayLogStream(tasks); err != nil {
+			return err
+		}
+	}
 	return nil
 }
 
+func (o *runTaskOpts) displayLogStream(tasks []*task.Task) error {
```

Comment:
Can we think about how we can move these methods to the `task` pkg instead of having the logic in the `cli` pkg?

For example, the interface could be:
```go
package task

type TaskDescriber interface {
     DescribeTasks(cluster string, taskARNs []string) ([]*ecs.Task, error)
}

type EventsWriter struct {
   GroupName string  
   Tasks []*Task
  
   // Fields that are private that get modified each time 
   lastEventTimestampByLogGroup map[string]int64
}

func (ew *EventsWriter) WriteEvents(w io.Writer) error {
   // Does not sleep and just writes the latest events to "w" when invoked.
   // updates the internal state of "ew" with the latest timestamps. 
}

func (ew *EventsWriter) WriteEventsUntilStopped(w io.Writer,  describer TaskDescriber) error {
   // Calls WriteEvents(w) in loop until all tasks are stopped.
}
```

Reply:
Hopefully that should also make unit testing easier :)